pyanaconda/ui/gui/spokes/installation_source.py

```diff
@@ -433,10 +433,17 @@ def __init__(self, *args, **kwargs):
         self._treeinfo_repos_already_disabled = False
 
     def apply(self):
-        payloadMgr.restart_thread(self.payload, checkmount=False)
+        source_changed = self._update_payload_source()
+        repo_changed = self._update_payload_repos()
+
+        if source_changed or repo_changed or self._error:
+            payloadMgr.restart_thread(self.payload, checkmount=False)
+        else:
+            log.debug("Nothing has changed - skipping payload restart.")
+
         self.clear_info()
 
-    def _method_changed(self):
+    def _update_payload_source(self):
         """ Check to see if the install method has changed.
```

Comment:
The naming doesn't exactly fit together, but I suppose that is for another time.